packages/flutter_test/lib/src/finders.dart

```diff
@@ -1026,6 +1066,69 @@ abstract class SemanticsFinder extends FinderBase<SemanticsNode> {
   }
 }
 
+/// A base class for creating finders that search for static text rendered by a
+/// [RenderParagraph].
+class _StaticTextRangeFinder extends _MapFinder<TextRangeContext, Element> {
+  /// Creates a new [SemanticsFinder] that will search starting at the given
+  /// `root`.
+  _StaticTextRangeFinder(this._parent);
+
+  @override
+  final _TextContainingWidgetFinder _parent;
+
+  @override
+  Iterable<TextRangeContext> _flatMap(Element from) {
+    final RenderObject? renderObject = from.renderObject;
+    // This is currenly only exposed on text matchers. Only consider RenderBoxes.
+    if (renderObject is! RenderBox) {
+      return const Iterable<TextRangeContext>.empty();
+    }
+
+    final View view = from.findAncestorWidgetOfExactType<View>()!;
+    final List<RenderParagraph> paragraphs = <RenderParagraph>[];
+
+    void visitor(RenderObject child) {
+      switch (child) {
+        case RenderParagraph():
+          paragraphs.add(child);
+          // No need to continue, we are piggybacking off of a text matcher, so
+          // inline text widgets will be reported separately.
+        case RenderBox():
+          child.visitChildren(visitor);
+        case _:
+      }
+    }
+    visitor(renderObject);
+    Iterable<TextRangeContext> searchInParagraph(RenderParagraph paragraph) {
+      final String text = paragraph.text.toPlainText();
+      return _parent.pattern.allMatches(text)
+        .map((Match match) => TextRangeContext._(view, paragraph, TextRange(start: match.start, end: match.end)));
+    }
+    return paragraphs.expand(searchInParagraph);
+  }
+
+  @override
+  String describeMatch(Plurality plurality) {
+    return switch (plurality) {
+      Plurality.zero || Plurality.many => 'non-overlapping TextRanges that match the Pattern "${_parent.pattern}"',
+      Plurality.one => 'non-overlapping TextRange that matches the Pattern "${_parent.pattern}"',
+    };
+  }
+}
+
+abstract class _MapFinder<To, From> extends FinderBase<To> {
```

Comment:
this only has one usage. Is it worthwhile to extract this abstraction as a concept right now or could this just be inlined in the only class that uses it?

Reply:
(I forgot to address this one I'll push a new commit in a minute).